### PR TITLE
API Group Overlap Validation Among Operators

### PR DESCRIPTION
**Pull request by orenc1**
A new verification is added during the process of build-manifests.sh, to check whether there are different operators in Kubevirt eco-system which are using the same API groups.
If one or more of such overlaps are detected, an error is shown and build-manifests.sh is exited.

Signed-off-by: orenc1 <ocohen@redhat.com>

```release-note
API groups overlap among different operators validation.
```

### CODE REVIEW

**Comment by orenc1** [comment]
**Note:** Currently the test fails because the API group `kubevirt.io` is being used by these operators: kubevirt, hco, node-maintenance

/cc @tiraboschi

File: tools/csv-merger/csv-merger.go [hunk 1]
```diff
@@ -89,6 +94,7 @@ var (
 		"Comma separated list of all the CRDs that should be visible in OLM console")
 	relatedImagesList = flag.String("related-images-list", "",
 		"Comma separated list of all the images referred in the CSV (just the image pull URLs or eventually a set of 'image|name' collations)")
+	crdDir              = flag.String("crds-dir", "", "the directory containing the CRDs for apigroup validation")
```
Comment (tiraboschi) [inline tools/csv-merger/csv-merger.go]: please modify the help message saying that the validation will be performed only if the value is != ""

Comment (orenc1) [reply tools/csv-merger/csv-merger.go]: Done, and I removed the triggering of the test from `build-manifests.sh`

**Pull request merged**